### PR TITLE
completed ticket 8

### PR DESCRIPTION
**Pull request by manheicatherine**

### CODE REVIEW

File: src/components/AddComments.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    if (newComment.username !== "") {
+      postArticleVote(article_id, newComment)
+        .then((res) => alert("Posted successfully. Please refresh the page."))
+        .catch((err) => {
+          alert("Please enter a correct username or valid comments. Please refresh the page.");
+        });
+      setIsPost(true);
+    }
+  }, [newComment, article_id]);
+
```
Comment (itskatherine) [inline src/components/AddComments.jsx:37]: Though you have posted the comment successfully to the database, it's not updating on the page (something which you're covering by asking the user to refresh the page)
However we can't be asking the user to refresh every time, what we need to do is use state to update the comments following a successful post request.
So we post the comment to the db, and then update the comments state so it includes the new posted comment
Can you please add this and then resubmit this PR

File: src/components/AddComments.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    body: "",
+  });
+
+  useEffect(() => {
+    setIsPost(false);
+    if (newComment.username !== "") {
+      postArticleVote(article_id, newComment)
+        .then((res) => alert("Posted successfully. Please refresh the page."))
+        .catch((err) => {
+          alert("Please enter a correct username or valid comments. Please refresh the page.");
```
Comment (itskatherine) [inline src/components/AddComments.jsx]: Try to avoid using alerts/pop ups, many browsers block them

File: src/components/AddComments.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        });
+      setIsPost(true);
```
Comment (itskatherine) [inline src/components/AddComments.jsx]: If we want to set isPosted to true after a post has happened successfully, this line should go in the .then block that follows a successful post

File: src/utils/api.js [hunk 4]
```diff
@@ -25,5 +25,10 @@ export function getCommentsOfArticle(id) {
 
 export function updateArticleVote(id, votes) {
   const patchBody = { inc_votes: votes };
-  return articlesApi.patch(`/articles/${id}`, patchBody)
+  return articlesApi.patch(`/articles/${id}`, patchBody);
+}
+
+export function postArticleVote(id, body) {
+  console.log(body)
```
Comment (itskatherine) [inline src/utils/api.js]: Remember to get rid of console logs for PRs 🙂

File: src/components/AddComments.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    let obj = {
+      username: e.target[0].value,
+      body: e.target[1].value,
+    };
+    setNewComment((current) => ({ ...current, ...obj }));
+  };
+
```
Comment (itskatherine) [inline src/components/AddComments.jsx:40]: The info that has been typed into your comment box is remaining after you press the submit button.
Remember than text input boxes are [controlled components](https://notes.northcoders.com/courses/js-front-end/controlled-components), and will need a state for the text being input into the form element, which will be transferred over to your newComment state on submitting.

**Review by itskatherine — CHANGES_REQUESTED** [review]
Hi Catherine, a great start but a few things to look at below 
Have a go at addressing my comments and submit another PR once you have please 🙂

**Review by elijaharley — APPROVED** [review]
Great stuff Catherine, the functionality is really nice on here now, and the comment displays immediately, giving feedback to the user that they have succeeded.

**Pull request merged**